doc/whatsnew/fragments/8433.user_action

```diff
@@ -0,0 +1,10 @@
+The following utilities are deprecated in favor of the more robust ``in_type_checking_block``
```

Comment:
I think it should be an internal change not user action. I think user action should be for user and not plugin maintainers.

Reply:
[f066909](https://github.com/PyCQA/pylint/pull/8433/commits/f0669098ea5055c328d9b5a31440d93efdea7404)